code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -179,6 +185,31 @@
 				beaker = null
 				cut_overlays()
 				. = TRUE
+		if("dispense_recipe")
+			var/recipe_to_use = params["recipe"]
+			var/list/chemicals_to_dispense = process_recipe_list(recipe_to_use)
+			for(var/r_id in chemicals_to_dispense) // i suppose you could edit the list locally before passing it
+				if(beaker && dispensable_reagents.Find(r_id)) // but since we verify we have the reagent, it'll be fine
+					var/amt = chemicals_to_dispense[r_id]
+					beaker.reagents.add_reagent(r_id, amt)
```

Comment:
lol you're not even using cell energy this is even better than i thought!

Reply:
thanks for pointing this out, now i'll fix it